pkg/cache/queue/cluster_queue.go

```diff
@@ -342,6 +371,7 @@ func (c *ClusterQueue) Pop() *workload.Info {
 	}
 
 	c.popCycle++
+	c.sw.clear()
```

Comment:
Actually, I'm not totally sure this is the right place to clear it. This is triggered on any call to Pop (nomination), so it wouldn't stick for longer than a single cycle. Note that a cycle might be triggered by inflow of new workloads unrelated to those.

Reply:
I see, that it is a valuable scenario to cover with a test then. Might be a follow up, but we need to make sure the investigation isn't lost.

As an alternative we could probably clear it inside the `RequeueIfNotPresent` if reason is different than `RequeueReasonPendingPreemption`, but I'm here, as long as we have another fix for this flake: https://github.com/kubernetes-sigs/kueue/pull/7157#discussion_r2402703554

Reply:
I added a test. With or without the line, the test passes, as the workload will go to the inadmissible queue. I worry that there could be a suble bug lurking here. However, I discussed offline with @mimiwo and we agreed that we shouldn't have a defensive LOC unless we can show it is problematic with a test.